pkg/providers/aws/provider_redis_test.go

```diff
@@ -604,7 +696,13 @@ func TestAWSRedisProvider_TagElasticache(t *testing.T) {
 			args: args{
 				ctx:      context.TODO(),
 				r:        buildTestRedisCR(),
-				cacheSvc: &mockElasticacheClient{replicationGroups: buildReplicationGroupReady()},
+				cacheSvc: buildMockElasticacheClient(func(elasticacheClient *mockElasticacheClient) {
+					elasticacheClient.describeCacheClustersFn = func(input *elasticache.DescribeCacheClustersInput) (*elasticache.DescribeCacheClustersOutput, error) {
+						return &elasticache.DescribeCacheClustersOutput{
+							CacheClusters: buildCacheClusterList(),
```

Comment:
Same as the others, if there is time, we should refactor this `buildCacheClusterList`